scripts/fbi/hate_crime/geo_id_resolver.py

```diff
@@ -0,0 +1,175 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Resolving place dcid given the Pub_agency_name, Agency_type_name, State_Abbr
+from FBI hate crime dataset
+
+"""
+
+import os
+import io
+import sys
+import csv
+
+from .manual_geocodes import MANUAL_COUNTY_MAP
+
+# Allows the following module imports to work when running as a script
+_SCRIPT_PATH = os.path.dirname(os.path.abspath(__file__))
+sys.path.append(os.path.join(_SCRIPT_PATH,
+                             '../../../util'))  # For State and County map
+from alpha2_to_dcid import USSTATE_MAP
+from county_to_dcid import COUNTY_MAP
+
+city_geocodes_csv_path = os.path.join(_SCRIPT_PATH,
+                                      '../crime/city_geocodes.csv')
+
+_US_GEO_CODE_UPDATE_MAP = {
+    # Replacing/Updating State abbreviations given in the data. Required in case of wrong state abbr mentioned in the data.
+    'NB': 'NE',  # Nebraska State
+    'GM': 'GU'  # Guam State
+}
+
+_IGNORE_STATE_ABBR = ['FS']  # Federal mentioned as State in data
+
+_IGNORE_COUNTY_SUFFIX = [
+    'Unified Police Department', 'Police Department', 'Public Safety'
+]
+
+city_geo_codes = {}
+with open(os.path.join(city_geocodes_csv_path), encoding="utf8") as csvfile:
+    csv_reader = csv.reader(csvfile)
+    for a in csv_reader:
+        city_geo_codes[a[0]] = a[1]
+
+
+def update_state_abbr(state_abbr):
```

Comment:
please add types for arguments and returned values for all functions, esp non-local function not beginning with '_':
Also add a block comment describing args and return values

def update_state_abbr(state_abbr: str) -> str:
    """
    Args:
       state_abbr: two letter code for a US state

   Returns:
       corrected two letter state code
    """

Reply:
Ack